internal/cli/cmd/dac/build/build.go

```diff
@@ -114,13 +119,21 @@ func (o *option) Execute() error {
 	return nil
 }
 
-func (o *option) processFile(file string) error {
-	// NB: most of the work of the `build` command is actually made by the `eval` command of the cue CLI.
-	// NB2: Since cue is written in Go, we could consider relying on its code instead of going the exec way.
-	//      However the cue code is (for now at least) not well packaged for such external reuse.
-	//      See https://github.com/cue-lang/cue/blob/master/cmd/cue/cmd/eval.go#L87
-	// NB3: #nosec is needed here even if the user-fed parts of the command are sanitized upstream
-	cmd := exec.Command("cue", "eval", file, "--out", o.Output, "--concrete") // #nosec
+func (o *option) processFile(file string, extension string) error {
+	var cmd *exec.Cmd
+
+	if extension == goExtension {
+		cmd = exec.Command("go", "run", file) // #nosec
+	} else if extension == cueExtension {
+		// NB: most of the work of the `build` command is actually made by the `eval` command of the cue CLI.
+		// NB2: Since cue is written in Go, we could consider relying on its code instead of going the exec way.
+		//      However the cue code is (for now at least) not well packaged for such external reuse.
+		//      See https://github.com/cue-lang/cue/blob/master/cmd/cue/cmd/eval.go#L87
+		// NB3: #nosec is needed here even if the user-fed parts of the command are sanitized upstream
+		cmd = exec.Command("cue", "eval", file, "--out", o.Output, "--concrete") // #nosec
+	} else {
+		return fmt.Errorf("invalid extension provided: must be either `cue` or `go`")
```

Comment:
we could just ignore a file that is not supported and instead of an error we can just log a warning

Reply:
Fine to me to keep returning err, method calling it handle the error as they want, logging or returning error

Reply:
the method calling this one cannot make the difference between the error from the command and a file not supported.

By the way: `go run `my_project/` can work too. You don't need to provide a specific file, you can provide a complete package and it will find for yourself the main file.

Reply:
Changed to warning.

For "go run folder/", I think it will hard to detect if the folder is containing go main exec without user passing the info via param. And this command it not walking in the repo as current implem. With current implem, you could target a folder containing multiple package having main file.